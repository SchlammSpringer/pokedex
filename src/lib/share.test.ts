import backImage from '$lib/assets/images/_gen/back'
import frontImage from '$lib/assets/images/_gen/front'
import habitatImage from '$lib/assets/images/_gen/habitats'
import classicImage from '$lib/assets/images/_gen/official-artwork'
import shinyImage from '$lib/assets/images/_gen/shiny'
import shinyBackImage from '$lib/assets/images/_gen/shinyback'
import { range } from 'fp-ts/NonEmptyArray'
import { describe, expect, it } from 'vitest'
import {
  getBackImage,
  getClassicSet,
  getFrontImage,
  getHabitatSet,
  getShinyBackImage,
  getShinyImage
} from './share'

describe('test image selecting', () => {
  describe('classic offical image', () => {
    it('all images present', () => {
      const pokedex = range(1, 151)
        .concat(10037)
        .map((it) => it.toString())

      const setSize = Object.keys(classicImage)

      expect(setSize).toEqual(pokedex)
    })
    it('find image in set', () => {
      const image = getClassicSet('1')

      expect(image).matchSnapshot()
    })
    it('default image if not in set', () => {
      const image = getClassicSet('abc')

      expect(image).toEqual(classicImage[10037])
    })
  })

  describe('habitat image', () => {
    it('all images present', () => {
      const habitat = [
        'cave',
        'forest',
        'grassland',
        'mountain',
        'rare',
        'rough-terrain',
        'sea',
        'urban',
        'waters-edge'
      ]

      const setSize = Object.keys(habitatImage)

      expect(setSize).toEqual(habitat)
    })
    it('find image in set', () => {
      const image = getHabitatSet('cave')

      expect(image).matchSnapshot()
    })
    it('default image if not in set', () => {
      const image = getHabitatSet('abc')

      expect(image).toEqual(getHabitatSet('cave'))
    })
  })

  describe('sprite image', () => {
    it('all images present', () => {
      const pokedex = range(1, 151)
        .map((it) => it.toString())
        .concat('egg')

      const frontSetSize = Object.keys(frontImage)
      const backSetSize = Object.keys(backImage)
      const shinySetSize = Object.keys(shinyImage)
      const shinyBackSetSize = Object.keys(shinyBackImage)

      expect(frontSetSize).toEqual(pokedex)
      expect(backSetSize).toEqual(pokedex)
      expect(shinySetSize).toEqual(pokedex)
      expect(shinyBackSetSize).toEqual(pokedex)
    })
    it('find image in set', () => {
      const frontalImage = getFrontImage('1')
      const backImage = getBackImage('1')
      const shinyImage = getShinyImage('1')
      const shinyBackImage = getShinyBackImage('1')

      expect(frontalImage).matchSnapshot()
      expect(backImage).matchSnapshot()
      expect(shinyImage).matchSnapshot()
      expect(shinyBackImage).matchSnapshot()
    })
    it('default image if not in set', () => {
      const front = getFrontImage('abc')
      const back = getBackImage('abc')
      const shiny = getShinyImage('abc')
      const shinyBack = getShinyBackImage('abc')

      expect(front).toEqual(frontImage['egg'])
      expect(back).toEqual(backImage['egg'])
      expect(shiny).toEqual(shinyImage['egg'])
      expect(shinyBack).toEqual(shinyBackImage['egg'])
    })
  })
})
