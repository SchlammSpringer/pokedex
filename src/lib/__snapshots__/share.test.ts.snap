// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test image selecting > classic offical image > find image in set 1`] = `
{
  "credit": null,
  "default": "\\\\assets\\\\pokemons\\\\official-artwork\\\\_gen\\\\1-384.webp?v=1682506628131",
  "formats": {
    "avif": {
      "180": "\\\\assets\\\\pokemons\\\\official-artwork\\\\_gen\\\\1-180.avif?v=1682506628131",
      "256": "\\\\assets\\\\pokemons\\\\official-artwork\\\\_gen\\\\1-256.avif?v=1682506628131",
      "310": "\\\\assets\\\\pokemons\\\\official-artwork\\\\_gen\\\\1-310.avif?v=1682506628131",
      "384": "\\\\assets\\\\pokemons\\\\official-artwork\\\\_gen\\\\1-384.avif?v=1682506628131",
      "475": "\\\\assets\\\\pokemons\\\\official-artwork\\\\_gen\\\\1-475.avif?v=1682506628131",
    },
    "webp": {
      "180": "\\\\assets\\\\pokemons\\\\official-artwork\\\\_gen\\\\1-180.webp?v=1682506628131",
      "256": "\\\\assets\\\\pokemons\\\\official-artwork\\\\_gen\\\\1-256.webp?v=1682506628131",
      "310": "\\\\assets\\\\pokemons\\\\official-artwork\\\\_gen\\\\1-310.webp?v=1682506628131",
      "384": "\\\\assets\\\\pokemons\\\\official-artwork\\\\_gen\\\\1-384.webp?v=1682506628131",
      "475": "\\\\assets\\\\pokemons\\\\official-artwork\\\\_gen\\\\1-475.webp?v=1682506628131",
    },
  },
  "placeholder": "data:image/webp;base64,UklGRvgAAABXRUJQVlA4WAoAAAAQAAAADwAADwAAQUxQSFQAAAARL0CkbZubnLHhK3hEBH0KtpFstXmgBrC5bO5zXwCSXv/FYGqI6P8EoFyZqF6uDdD4yA34bpIUislqTPk9EzJHSqo/IfAmJMLzJ/Zh50wBJVSnDPJWUDggfgAAAHACAJ0BKhAAEAAFQHwlsAJ0MFERoGzaSmE38AAA+SxPtHn+oVWFXjP7St3oRjnHK4+dVw58VUFxlRnRZnjhm0gT11KyqyULTnma/PY9GTWc0aAVfixvivbkKEO85C7eWi89bA1FpGXnwElVOVzg97NpJaadOymxmhY9QAYAAA==",
  "sizes": {
    "180": {
      "avif": "\\\\assets\\\\pokemons\\\\official-artwork\\\\_gen\\\\1-180.avif?v=1682506628131",
      "webp": "\\\\assets\\\\pokemons\\\\official-artwork\\\\_gen\\\\1-180.webp?v=1682506628131",
    },
    "256": {
      "avif": "\\\\assets\\\\pokemons\\\\official-artwork\\\\_gen\\\\1-256.avif?v=1682506628131",
      "webp": "\\\\assets\\\\pokemons\\\\official-artwork\\\\_gen\\\\1-256.webp?v=1682506628131",
    },
    "310": {
      "avif": "\\\\assets\\\\pokemons\\\\official-artwork\\\\_gen\\\\1-310.avif?v=1682506628131",
      "webp": "\\\\assets\\\\pokemons\\\\official-artwork\\\\_gen\\\\1-310.webp?v=1682506628131",
    },
    "384": {
      "avif": "\\\\assets\\\\pokemons\\\\official-artwork\\\\_gen\\\\1-384.avif?v=1682506628131",
      "webp": "\\\\assets\\\\pokemons\\\\official-artwork\\\\_gen\\\\1-384.webp?v=1682506628131",
    },
    "475": {
      "avif": "\\\\assets\\\\pokemons\\\\official-artwork\\\\_gen\\\\1-475.avif?v=1682506628131",
      "webp": "\\\\assets\\\\pokemons\\\\official-artwork\\\\_gen\\\\1-475.webp?v=1682506628131",
    },
  },
}
`;

exports[`test image selecting > habitat image > find image in set 1`] = `
{
  "credit": null,
  "default": "\\\\assets\\\\habitats\\\\habitats\\\\_gen\\\\cave-1024.webp?v=1682065423028",
  "formats": {
    "avif": {
      "1024": "\\\\assets\\\\habitats\\\\habitats\\\\_gen\\\\cave-1024.avif?v=1682065423028",
      "384": "\\\\assets\\\\habitats\\\\habitats\\\\_gen\\\\cave-384.avif?v=1682065423028",
      "640": "\\\\assets\\\\habitats\\\\habitats\\\\_gen\\\\cave-640.avif?v=1682065423028",
      "768": "\\\\assets\\\\habitats\\\\habitats\\\\_gen\\\\cave-768.avif?v=1682065423028",
    },
    "webp": {
      "1024": "\\\\assets\\\\habitats\\\\habitats\\\\_gen\\\\cave-1024.webp?v=1682065423028",
      "384": "\\\\assets\\\\habitats\\\\habitats\\\\_gen\\\\cave-384.webp?v=1682065423028",
      "640": "\\\\assets\\\\habitats\\\\habitats\\\\_gen\\\\cave-640.webp?v=1682065423028",
      "768": "\\\\assets\\\\habitats\\\\habitats\\\\_gen\\\\cave-768.webp?v=1682065423028",
    },
  },
  "placeholder": "data:image/webp;base64,UklGRjwAAABXRUJQVlA4IDAAAADwAQCdASoQAAYABUB8JQBOgCIL23gBuoAA/tyT2qIllwMzAMKQxU5iPs00hZow0AA=",
  "sizes": {
    "1024": {
      "avif": "\\\\assets\\\\habitats\\\\habitats\\\\_gen\\\\cave-1024.avif?v=1682065423028",
      "webp": "\\\\assets\\\\habitats\\\\habitats\\\\_gen\\\\cave-1024.webp?v=1682065423028",
    },
    "384": {
      "avif": "\\\\assets\\\\habitats\\\\habitats\\\\_gen\\\\cave-384.avif?v=1682065423028",
      "webp": "\\\\assets\\\\habitats\\\\habitats\\\\_gen\\\\cave-384.webp?v=1682065423028",
    },
    "640": {
      "avif": "\\\\assets\\\\habitats\\\\habitats\\\\_gen\\\\cave-640.avif?v=1682065423028",
      "webp": "\\\\assets\\\\habitats\\\\habitats\\\\_gen\\\\cave-640.webp?v=1682065423028",
    },
    "768": {
      "avif": "\\\\assets\\\\habitats\\\\habitats\\\\_gen\\\\cave-768.avif?v=1682065423028",
      "webp": "\\\\assets\\\\habitats\\\\habitats\\\\_gen\\\\cave-768.webp?v=1682065423028",
    },
  },
}
`;

exports[`test image selecting > sprite image > find image in set 1`] = `
{
  "credit": null,
  "default": "\\\\assets\\\\sprites\\\\front\\\\_gen\\\\1-96.webp?v=1682519026035",
  "formats": {
    "avif": {
      "96": "\\\\assets\\\\sprites\\\\front\\\\_gen\\\\1-96.avif?v=1682519026035",
    },
    "webp": {
      "96": "\\\\assets\\\\sprites\\\\front\\\\_gen\\\\1-96.webp?v=1682519026035",
    },
  },
  "placeholder": "data:image/webp;base64,UklGRroAAABXRUJQVlA4WAoAAAAQAAAADwAADwAAQUxQSDsAAAARL6AmAAE2p4t7fSqIiMDUE4oiSWruUAAOwEEUkAWk9S/meKEgov8TQD/7/CgAiwUoUQEWdt6HdoZ+BgBWUDggWAAAABACAJ0BKhAAEAAFQHwlsAJ0fwAO0N6mjcAA/FsaF/nNoYk9VF1YDZQvwqpI/ACdsWUFQaO3FAL5J6h1QegHxaJTEiS3elnujgUFrPqM3fTOgma0gPMAAAA=",
  "sizes": {
    "96": {
      "avif": "\\\\assets\\\\sprites\\\\front\\\\_gen\\\\1-96.avif?v=1682519026035",
      "webp": "\\\\assets\\\\sprites\\\\front\\\\_gen\\\\1-96.webp?v=1682519026035",
    },
  },
}
`;

exports[`test image selecting > sprite image > find image in set 2`] = `
{
  "credit": null,
  "default": "\\\\assets\\\\sprites\\\\back\\\\_gen\\\\1-96.webp?v=1682519026035",
  "formats": {
    "avif": {
      "96": "\\\\assets\\\\sprites\\\\back\\\\_gen\\\\1-96.avif?v=1682519026035",
    },
    "webp": {
      "96": "\\\\assets\\\\sprites\\\\back\\\\_gen\\\\1-96.webp?v=1682519026035",
    },
  },
  "placeholder": "data:image/webp;base64,UklGRq4AAABXRUJQVlA4WAoAAAAQAAAADwAADwAAQUxQSDUAAAARL6AmABI2SBYCcOI2IgLy8KEgsk0LCYwIEghBhNs/DO+bIKL/E0A/26RIbgHIgqeirm7DrwBWUDggUgAAABACAJ0BKhAAEAAFQHwlsALD4EABMWSLshAA7UdXs4mH/IDZ15oGyA5RcMy1DQj85uXYM/uR/FzUxklGMfdSSBEAt7tnwcbDIiBZOTptCdY6wAA=",
  "sizes": {
    "96": {
      "avif": "\\\\assets\\\\sprites\\\\back\\\\_gen\\\\1-96.avif?v=1682519026035",
      "webp": "\\\\assets\\\\sprites\\\\back\\\\_gen\\\\1-96.webp?v=1682519026035",
    },
  },
}
`;

exports[`test image selecting > sprite image > find image in set 3`] = `
{
  "credit": null,
  "default": "\\\\assets\\\\sprites\\\\shiny\\\\_gen\\\\1-96.webp?v=1682519026035",
  "formats": {
    "avif": {
      "96": "\\\\assets\\\\sprites\\\\shiny\\\\_gen\\\\1-96.avif?v=1682519026035",
    },
    "webp": {
      "96": "\\\\assets\\\\sprites\\\\shiny\\\\_gen\\\\1-96.webp?v=1682519026035",
    },
  },
  "placeholder": "data:image/webp;base64,UklGRrwAAABXRUJQVlA4WAoAAAAQAAAADwAADwAAQUxQSDsAAAARL6AmAAE2p4t7fSqIiMDUE4oiSWruUAAOwEEUkAWk9S/meKEgov8TQD/7/CgAiwUoUQEWdt6HdoZ+BgBWUDggWgAAAJACAJ0BKhAAEAAFQHwlsAJ0fwYn/ZFBEPFk8Xw4APxaa6AsPslD0kP6pSoQ64rdcbh7jVYpY5F4ZVbNK/YZxadsYZPSLpBt7Xq7oXubbxbThuNAp3nKY9fAAA==",
  "sizes": {
    "96": {
      "avif": "\\\\assets\\\\sprites\\\\shiny\\\\_gen\\\\1-96.avif?v=1682519026035",
      "webp": "\\\\assets\\\\sprites\\\\shiny\\\\_gen\\\\1-96.webp?v=1682519026035",
    },
  },
}
`;

exports[`test image selecting > sprite image > find image in set 4`] = `
{
  "credit": null,
  "default": "\\\\assets\\\\sprites\\\\shinyback\\\\_gen\\\\1-96.webp?v=1682519026035",
  "formats": {
    "avif": {
      "96": "\\\\assets\\\\sprites\\\\shinyback\\\\_gen\\\\1-96.avif?v=1682519026035",
    },
    "webp": {
      "96": "\\\\assets\\\\sprites\\\\shinyback\\\\_gen\\\\1-96.webp?v=1682519026035",
    },
  },
  "placeholder": "data:image/webp;base64,UklGRrYAAABXRUJQVlA4WAoAAAAQAAAADwAADwAAQUxQSDUAAAARL6AmABI2SBYCcOI2IgLy8KEgsk0LCYwIEghBhNs/DO+bIKL/E0A/26RIbgHIgqeirm7DrwBWUDggWgAAAFACAJ0BKhAAEAAFQHwlsAJ0fwPYtBik53kyAAD+ucNhWhawG4K6VIQottNEnTzt1eUkpdUsNMthJRL1l2x6GBZd/au2F4EgMUVAgNxc3jhVntVQLbmF5wAAAA==",
  "sizes": {
    "96": {
      "avif": "\\\\assets\\\\sprites\\\\shinyback\\\\_gen\\\\1-96.avif?v=1682519026035",
      "webp": "\\\\assets\\\\sprites\\\\shinyback\\\\_gen\\\\1-96.webp?v=1682519026035",
    },
  },
}
`;
